### PR TITLE
feat: Do not add hash in test environment

### LINKED ISSUE #870: Way to disable hash in test environment?
Would it be nice if we could disable hash className and only use the `label` in test environment?

### PR DESCRIPTION
**Pull request by huchenme**
This PR Closes #870 #869

**What**: 

in TEST environment:

```
const cls1 = css`
 font-size: 20px;
 background: green;
 label: label1;
`

const cls2 = css`
 font-size: 20px;
 background: red;
 label: label2;
`

const cn = cx(cls1, cls2, 'other')
```

Before:

```
console.log(cls1) // css-12345-label1
console.log(cls2) // css-12345-label2
console.log(cn) // other css-12345-label1-label2
```

After:

```
console.log(cls1) // css-label1
console.log(cls2) // css-label2
console.log(cn) // label1 label2 other css-12345-label1-label2
```

**Why**:

With this feature, it is easier to use enzyme CSS selectors

**How**:

* If `process.env.NODE_ENV === 'test'` and `label` is not empty, disable hash.
* Do not disable hash for `injectGlobal` as it will break the `emotion-server` tests
* Add className without key to `rawClassName` list (it will be problematic with keys)

**Checklist**:

- [ ] Documentation (any place I should add this?)
- [x] Tests
- [x] Code complete

### CODE REVIEW

File: packages/emotion-server/test/auto-label/__snapshots__/auto-label.test.js.snap [hunk 1]
```diff
@@ -2,15 +2,7 @@
 
 exports[`extractCritical returns static css 1`] = `
 Object {
-  "css": "@font-face {
```
Comment (huchenme) [inline packages/emotion-server/test/auto-label/__snapshots__/auto-label.test.js.snap]: I might need some help with this, tried for a few hours still no luck to get `injectGlobal` work

**Comment by huchenme** [comment]
Could someone help review, please? @tkh44 @mitchellhamilton

**Comment by mitchellhamilton** [comment]
Thanks for the PR but this would break if two labels are the same so I don't think this is something we can accept. For your original issues, I would recommend adding something like a `data-testid` attribute to the elements that you want to select.

**Comment by mitchellhamilton** [comment]
@huchenme How would that work?

The whole point is that using a hash makes each class name unique and deterministic, how would you maintain the uniqueness and determinism without using hashes.

**Comment by huchenme** [comment]
It is important that in Jest tests, people are able to select by classNames. That is why Jest have mentioned on [their website](https://jestjs.io/docs/en/webpack#mocking-css-modules) and recommended to use [identity-obj-proxy](https://github.com/keyanzhang/identity-obj-proxy).

I want to achieve the same in the PR, why do we need to keep className unique in TEST environment?

**Comment by Andarist** [comment]
I agree with @mitchellhamilton that what u actually might want to use are `data-testid`s, they could also be useful in your E2E tests (where u wont be able to target elements by classnames anyway)

**Comment by mitchellhamilton** [comment]
> why do we need to keep className unique in TEST environment?

Because there could still be clashes.

**Pull request closed without merging**